### PR TITLE
BXMSDOC-3619: Modularizing Using Ansible & RHDM in ONAP document

### PR DESCRIPTION
**Pull request by gadeyredhat**
[Jira ID](https://issues.jboss.org/browse/BXMSDOC-3619)
[DM Rendered Output](http://file.pnq.redhat.com/~gadey/BXMSDOC-3619-dm/)

[Original Google Doc](https://docs.google.com/document/d/1G6mSPasP6dCwI-_u996GQgEBk342LJGcyBhov7Alqqo/edit?pli=1)

### CODE REVIEW

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/technology-and-definitions-con.adoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[id='technology-and-definitions-con']
+= Technology and definitions
+
+== What is ONAP?
+
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/technology-and-definitions-con.adoc:5]: For the sake of being consistent with our other user story types, can we not have questions as titles? (applies to all such titles in this doc)

File: assemblies/assembly_using-ansible-rhdm-in-ONAP/dm/master-docinfo.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<productname>{PRODUCT}</productname>
+<productnumber>{PRODUCT_VERSION}</productnumber>
```
Comment (vidya-iyengar) [inline assemblies/assembly_using-ansible-rhdm-in-ONAP/dm/master-docinfo.xml]: As discussed, there should be no occurrence of the word "Drools" throughout the doc including the title. Please make the global changes.

File: assemblies/assembly_using-ansible-rhdm-in-ONAP/main.adoc [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+= Policy based decision and control in ONAP using Ansible and {PRODUCT}
+// Context attribute is assembly specific and enables module reuse between assemblies.
+:doctype: book
+:imagesdir: _images
+:context: using-ansible-rhdm-in-ONAP
+include::_artifacts/author-group.adoc[]
+
+// Purpose statement for the assembly
+This solution examines the use of Ansible and {PRODUCT} technology in ONAP architecture. It elaborates on how automation and rules management systems together can provide a comprehensive network policy engine capable of providing the necessary automations for proactive response to network and service conditions without human intervention which is also known as closed loop automation.
```
Comment (vidya-iyengar) [inline assemblies/assembly_using-ansible-rhdm-in-ONAP/main.adoc]: The sentence is too long and incomprehensible. Please consider splitting into smaller ones.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
+[id='rules-engines-and-rhdm-con']
+= Rules Engines and {PRODUCT}
+
+A rules engine is a software element that makes decisions based on business logic built using declarative format. Rules engine can be very basic providing support for simple rules to very complicated based on sophisticated algorithms.
+
+{PRODUCT} provides a versatile set of decision capabilities, including a rules engine based on the link:https://access.redhat.com/documentation/en-us/red_hat_jboss_bpm_suite/6.2/html/development_guide/rule_evaluation_with_phreak_algorithm[PHREAK] algorithm (an evolution of the well know Rete algorithm), a link:https://www.omg.org/spec/DMN/About-DMN/[DMN v1.1 FEEL Compliance Level 3] runtime, support for link:https://www.packtpub.com/mapt/book/networking_and_servers/9781783288625/7/ch07lvl1sec47/pmml[PMML], as well as a Complex Event Processing engine.
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc]: We are pointing to the 6.2 version of the dev guide here. We do not support 6.2 version and pointing to the previous version of the docs is not ideal. So please remove those references to the earlier version of the product docs.
Also, regarding the two other references (to https://www.omg.org/spec/DMN/About-DMN/ and https://subscription.packtpub.com/book/networking_and_servers/9781783288625/7/ch07lvl1sec47/pmml), I'd prefer if we take Christy's ack on pointing to these resources from our doc.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+The PHREAK reasoning algorithm provides a runtime that easily scales to hundreds of thousands of rules in a single rules execution environment, while providing low-latency and high-performance business rules execution.
+
+Being a lightweight engine, {PRODUCT} can be used as the decision service and runtime in various architectures and deployment topologies, including, but not limited, embedded systems and microservices architectures. The decision engine can be combined with multiple application frameworks and runtimes.
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc]: "but not limited, " > "but not limited to, "

Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc]: "can be used" - lots of occurrences of passive voice usage throughout the doc.
For example:
"The decision engine can be combined with multiple application frameworks and runtimes. "
> " You can combine the decision engine with multiple application frameworks and runtimes. "

Please check for passive voice globally.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+{PRODUCT} is build on open standards (DMN, XML, JSON, JAX-RS) and de-facto standards (Git, Maven, Eclipse, IntelliJ).
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc]: build > built

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+{PRODUCT} is supported by Red Hat and is an equivalent of the open-source community project, Drools. {PRODUCT} has the following capabilities:
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc]: The sentence "Red Hat Decision Manager is supported by Red Hat and is an equivalent of the open-source community project, Drools." is not required. 
You can start this paragraph with  "Red Hat Decision Manager has the following capabilities" directly.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/automation-and-ansible-con.adoc [hunk 8]
```diff
@@ -0,0 +1,13 @@
+[id='automation-and-ansible-con']
+= Automation and Ansible
+
+Ansible is a radically simple automation engine that automates link:https://www.ansible.com/integrations/networks[network functions], link:https://www.ansible.com/provisioning?hsLang=en-us[cloud provisioning], link:https://www.ansible.com/configuration-management?hsLang=en-us[configuration management], link:https://www.ansible.com/application-deployment?hsLang=en-us[application deployment], link:https://www.ansible.com/orchestration?hsLang=en-us[intra-service orchestration], and many other automation needs.
+
+Designed for multi-tier deployments, Ansible models your environment by describing how all of your systems interrelate, rather than just managing one system at a time.
+
+It uses no agents and no additional custom security infrastructure, so it's easy to deploy - and most importantly, it uses a very simple language (YAML, in the form of Ansible Playbooks) that allow you to describe your automation jobs in a way that approaches plain English.
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/automation-and-ansible-con.adoc]: " so it’s easy to deploy - and most "
Don't use hyphen, instead split the sentence.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc [hunk 9]
```diff
@@ -0,0 +1,50 @@
+[id='policy-with-rhdm-con']
+= Policy with {PRODUCT}
+
+.ONAP policy creation and deployment
+image::using-ansible-rhdm-in-ONAP/ONAP-policy-create-deploy.png[ONAP policy creation and deployment]
+
+ONAP policy framework (referred as “POLICY”) is composed of following subcomponents:
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc]: (referred as “POLICY”)  >> (POLICY).

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+* *Policy Administration Point* (*PAP*): PAP offers interfaces for policy creation. The interfaces are integrated with a portal to provide a GUI.
+
+* *Policy Decision Point* (*PDP*) is based on a specific rules technology. It has two components:
+
+** *PDP-X* is based on XACML technology.  PDP-X is _stateless_ and can be deployed as a resource pool of PDP-X servers.  The number of servers can be grown to increase both capacity (horizontal scalability) and to increase availability.
+
+** *PDP-D* is _stateful_, as it utilizes *{PRODUCT}* in its native, stateful way and transactions persist so long as the PDP-D is active. It provides capability for advanced control loops, provides interfaces for various ONAP components to trigger actions and receive events and maintains state throughout the workflows across the network while handling failures in the corrective actions.
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc]: why is the product name in bold here?

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+After a policy has been initially created or an existing policy has been modified, the Policy Distribution Framework sends the policy from the repository to its points of use, which include Policy Decision Points (PDPs) and Policy enforcement points before the policy is actually needed.
+
+Examples of policy enforcement may include the following:
+
+* Policy rules for data collection and retention by DCAE data collection functionality.
+* Analytic policy rules, identification of anomalous or abnormal conditions, and publication of events signaling detection of such conditions by DCAE analytic applications.
+* Policy rules for associated remedial actions, or for further diagnostics, are enforced by the correct component in a control loop such as the MSO, a Controller, or DCAE.
+* Policy engines such as XACML and {PRODUCT} also enforce policies and can trigger other components as a result (for example, causing a controller to take specific actions specified by the policy).  Additionally, some policies (“Guard Policies”) may enforce checks against decided actions.
+
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc:47]: (“Guard Policies”) > (Guard Policies) 
Please check for similar occurrences throughout the docs.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/closed-loop-OPNFV-components-con.adoc [hunk 12]
```diff
@@ -0,0 +1,10 @@
+[id='closed-loop-OPNFV-components-con']
+= Closed loop with OPNFV components
+
+link:https://www.google.com/url?q=https://wiki.opnfv.org/display/fastpath/Barometer+Home[Barometer] is an OPNFV project that develops features in `collectd` to collect metrics and events suitable for NFV deployments.
```
Comment (vidya-iyengar) [inline doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/closed-loop-OPNFV-components-con.adoc]: The google link provided here is not yielding appropriate results. Please ask the sme what needs to be linked here.
Also,check for similar references to google results in the doc if any), those needs to be removed and replaced with correct and authentic references.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/technology-and-definitions-con.adoc (hunk 1 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/technology-and-definitions-con.adoc:5]: Changed as suggested. 

There were two occurrences of questions as titles, changed them both to:
- What is ONAP? to Open Network Automation Platform,
- What is policy? to Policy.

File: assemblies/assembly_using-ansible-rhdm-in-ONAP/main.adoc (hunk 3 above)
Comment (gadeyredhat) [reply assemblies/assembly_using-ansible-rhdm-in-ONAP/main.adoc]: Changed as suggested.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/closed-loop-OPNFV-components-con.adoc (hunk 12 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/closed-loop-OPNFV-components-con.adoc]: Link corrected. Removed the google link and placed the landing page of the Barometer project as suggested by the SME.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc (hunk 11 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc:47]: Changed as suggested. 
Found one more occurrence of double quotes (""), removed them.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc (hunk 10 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc]: Bold removed. In the original google doc, Drools is in bold, looks like the author wanted to emphasize on Drools/RHDM (our product). Hence I kept it as bold.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc (hunk 9 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/policy-with-rhdm-con.adoc]: Changed as suggested.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/automation-and-ansible-con.adoc (hunk 8 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/automation-and-ansible-con.adoc]: Split the sentence in two as suggested.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc (hunk 7 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc]: Changed as suggested.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc (hunk 6 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc]: Changed as suggested.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc (hunk 5 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc]: Changed as suggested.

File: doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc (hunk 4 above)
Comment (gadeyredhat) [reply doc-content/enterprise-only/using-ansible-rhdm-in-ONAP/rules-engines-and-rhdm-con.adoc]: I have updated the link to point to the 6.4 version of the dev guide which we are supporting now. 

Hi @Christywatkins, can we point to the following resources from this doc:
-  https://www.omg.org/spec/DMN/About-DMN/ and 
- https://subscription.packtpub.com/book/networking_and_servers/9781783288625/7/ch07lvl1sec47/pmml

File: assemblies/assembly_using-ansible-rhdm-in-ONAP/dm/master-docinfo.xml [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+<subtitle>
+</subtitle>
+<abstract>
+	<para>This paper examines the use of Ansible and {PRODUCT} technology in ONAP architecture.</para>
```
Comment (baldimir) [inline assemblies/assembly_using-ansible-rhdm-in-ONAP/dm/master-docinfo.xml]: "paper" sounds more like a scientific paper. Would it be possible to use the word "document" here instead, please?

**Review by baldimir — COMMENTED** [review]
Just one minor comment. Otherwise looks good, thanks!

File: assemblies/assembly_using-ansible-rhdm-in-ONAP/dm/master-docinfo.xml (hunk 13 above)
Comment (gadeyredhat) [reply assemblies/assembly_using-ansible-rhdm-in-ONAP/dm/master-docinfo.xml]: Changed as suggested.

**Review by baldimir — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Pull request merged**